lib/kafka/consumer.rb

```diff
@@ -103,6 +103,21 @@ def subscribe(topic, default_offset: nil, start_from_beginning: true, max_bytes_
       nil
     end
 
+    def subscribe_by_regex(topic_regex, default_offset: nil, start_from_beginning: true, max_bytes_per_partition: 1048576)
```

Comment:
Could we call this e.g. `subscribe_matching` instead?

Reply:
Or maybe just merge this into `subscribe`? I actually think it's better to just check if the user passes a regex there. Then existing integrations can just start passing regexes.